design/bios-config.md

```diff
@@ -0,0 +1,151 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# bios-config
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-15T11:41-05:00 -->
+
+<!--te-->
+
+## Summary
+
+This document explains the usage of YAML attributes for BIOS configuration for the BareMetalHost of the BMO. These configurations will be applied to the host according to the vendor's BMC driver being used by Ironic.
+
+## Motivation
+
+We need this document to specify the typical usage of BIOS YAML attributes in the BareMetalHost YAML. 
+
+
+### Goals
+
+1. To agree on the format of the attributes that are to be supported as a starting point
+2. To keep the attribute naming vendor agnostic
+
+### Non-Goals
+
+1. To list every attribute for every vendor exhaustively
+
+## Proposal
+The proposal is to set/unset the BIOS configuration values using vendor drivers available for each BMC type. For this purpose, new YAML attributes are to be introduced in the BareMetalHost spec. 
+ 
+### Implementation Details/Notes/Constraints 
+All the BIOS related attributes or fields will come under the sub-section called ```bios``` in spec. The values for each attribute can be a boolean or a string. The vendor driver type to implement this configuration is not to be specified separetely as it is already known from the bmc sub-section.
+Initially, the BIOS configuration can be applied using the controller types idrac and irmc
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+ name: bm0
+spec:
+ online: true
+ bmc:
+   address: idrac://192.168.122.1:6230/
+   credentialsName: bm0-bmc-secret
+ bootMACAddress: 52:54:00:b7:b2:6f	
+ bios:
+   EnableSriovGlobal: false
```

Comment:
We're going to need to ensure that the booleans are implemented as pointers so that we can tell whether the user is actually asking us to set a specific value.

Reply:
To allow them to set as null. Yes, makes sense.